tabsie/server.R

```diff
@@ -80,17 +80,48 @@ shinyServer(
     })
 ####### BUTTON PRESSES #####################
     observeEvent(input$clearTheme, {
-      if (!valAuth) return;#break processing of not authorized.
+      #if (!valAuth) return;#break processing of not authorized.
       updateTextInput(session, "titleField", value = "")
       updateTextInput(session, "xLab", value = "")
       updateTextInput(session, "yLab", value = "")
       updateSliderInput(session, "textSize", value = 15)
       updateSliderInput(session, "xLabRotation", value = 0)
       updateSliderInput(session, "xLabHeight", value = 0)
     })
+######## DIV BOX CONTROL for IDEAS and BUGS #################
+    observe({
+      #if (!valAuth) return();#break processing of not authorized.
+      shinyjs::onclick("toggleIdea",toggle(id="ideaDiv",anim=T))
+      shinyjs::onclick("toggleBug",toggle(id="bugDiv",anim=T))
+    })
+
+    observe({
+      #if (!valAuth) return();#break processing of not authorized.
+      needIdea <- need(input$ideaInput,F);
+      needBug <- need(input$bugInput,F);
+      if(length(c(needIdea,needBug))<=1) show('emailDiv',anim=T);
+      #browser();
+      if(is.null(needIdea)) show('submitIdea') else hide('submitIdea');
+      if(is.null(needBug)) show('submitBug') else hide('submitBug');
+    })
+    
+    observeEvent(input$submitIdea,{
+      #if (!valAuth) return();#break processing of not authorized.
+      updateTextInput(session,'ideaInput', value='');
+      hide('submitIdea');
+      hide(id="ideaDiv",anim=T);
+    })
+    
+    observeEvent(input$submitBug,{
+      #if (!valAuth) return();#break processing of not authorized.
+      updateTextInput(session,'bugInput', value='');
+      hide('submitBug');
+      hide(id="bugDiv",anim=T);
+    })
+    
```

Comment:
The above two are listeners for submission of either ideas or bugs. Programmatically all they do is empty the fields and collapse them, relying on the separate logging functionality to have captured their contents. But we need _something_ to happen when you press a submit button, or the users will get confused.